### PR TITLE
Add a test to verify select_console fails on broken vnc

### PR DESCRIPTION
**Pull request by coolo**

### CODE REVIEW

**Comment by okurz** [comment]
hah, told you so! ;) Actually I did not but that test proves the problem in a very good way. Thanks for this

File: consoles/vnc_base.pm [hunk 1]
```diff
@@ -35,8 +35,15 @@ sub screen {
 
 sub activate {
     my ($self) = @_;
-    $self->connect_vnc($self->{args});
-    return $self->SUPER::activate;
+    try {
+        local $SIG{__DIE__} = undef;
```
Comment (okurz) [inline consoles/vnc_base.pm]: ah. This is the part we missed. Something learned :)

**Comment by okurz** [comment]
LGTM although I would be interested in how it looks like in an openQA test. The travis CI test failed on setting up the initial ubuntu repos.

**Comment by Soulofdestiny** [comment]
seems to work :)

```
/home/os-autoinst/consoles/vnc_base.pm:71:{
  'port' => 5901,
  'password' => 'nots3cr3t',
  'hostname' => '10.161.145.8'
}
06:17:09.9962 7699 Error connecting to host <10.161.145.8>: IO::Socket::INET: connect: Connection refused
06:17:10.9972 7699 Error connecting to host <10.161.145.8>: IO::Socket::INET: connect: Connection refused
06:17:11.9982 7699 Error connecting to host <10.161.145.8>: IO::Socket::INET: connect: Connection refused
06:17:12.9992 7699 Error connecting to host <10.161.145.8>: IO::Socket::INET: connect: Connection refused
06:17:14.0002 7699 Error connecting to host <10.161.145.8>: IO::Socket::INET: connect: Connection refused
06:17:15.0010 7699 Error connecting to host <10.161.145.8>: IO::Socket::INET: connect: Connection refused
06:17:16.0019 7699 Error connecting to host <10.161.145.8>: IO::Socket::INET: connect: Connection refused
06:17:17.0028 7699 Error connecting to host <10.161.145.8>: IO::Socket::INET: connect: Connection refused
06:17:18.1312 7692 # Test died: Error connecting to host <10.161.145.8>: IO::Socket::INET: connect: Connection refused at /home/os-autoinst/testapi.pm line 1351.

06:17:18.1330 7684 awaiting death of testpid 7692
06:17:18.1367 7684 test process exited: 7692
06:17:18.1367 7684 awaiting death of commands process
06:17:18.1385 7684 commands process exited: 7686
06:17:18.1386 7684 isotovideo done
06:17:18.2230 7699 Connection to root@s390pb.suse.de established
06:17:18.3163 7699 Command executed: ! virsh dominfo openQA-SUT-22 | grep -w 'shut off', ret=0
06:17:18.3166 7684 BACKEND SHUTDOWN 0
06:17:18.3173 7699 Destroying openQA-SUT-22 virtual machine
06:17:18.4135 7699 Connection to root@s390pb.suse.de established
06:17:18.7322 7699 Command's stdout:
Domain openQA-SUT-22 destroyed


06:17:18.7324 7699 Command executed: virsh destroy openQA-SUT-22, ret=0
06:17:18.9084 7699 Connection to root@s390pb.suse.de established
06:17:19.0077 7699 Command's stdout:
Domain openQA-SUT-22 has been undefined


06:17:19.0079 7699 Command executed: virsh undefine openQA-SUT-22, ret=0
last frame
06:17:19.0960 7699 sending magic and exit
06:17:19.0962 7684 received magic close
XIO:  fatal IO error 11 (Resource temporarily unavailable) on X server ":54275"
XIO:  fatal IO error 11 (Resource temporarily unavailable) on X server ":42268"
      after 8706 requests (8706 known processed) with 0 events remaining.
      after 2679 requests (2679 known processed) with 0 events remaining.
7684: EXIT 0

```

**Comment by Soulofdestiny** [comment]
also looks good in the webui:
http://opeth/tests/5725#step/fail_early/2

**Pull request merged**